### PR TITLE
How to document different licence e.g. GPL

### PR DESCRIPTION
**Pull request by PallHaraldsson**

### CODE REVIEW

**Comment by PallHaraldsson** [comment]
[noblock] since AutoClosing doesn't apply here (nor AutoMerge). I should maybe have opened this as an issue, since so far no discussion or disagreement...

**Comment by DilumAluthge** [comment]
Maybe post in the #pkg-registration channel on Slack? The General registry has a very high rate of GitHub notifications, so these things can get missed.

**Comment by odow** [comment]
One example is ECOS: https://github.com/jump-dev/ECOS.jl/blob/master/LICENSE.md. We include a sentence at the bottom of the LICENSE, but open to suggestions for how this could be improved.

File: README.md [hunk 1]
```diff
@@ -48,8 +48,11 @@ repository after the new release of your package is merged into the registry.
 
 Registered packages must have an [Open Source Initiative approved license](https://opensource.org/licenses),
 clearly marked via a `LICENSE.md`, `LICENSE`, `COPYING` or similarly named file in the package repository.
-Packages that wrap proprietary libraries are acceptable if the licenses of those libraries permit open
-source distribution of the Julia wrapper code.
+Packages that wrap proprietary libraries (or otherwise restrictive libraries, such as copyleft GPL code) are
```
Comment (aviks) [inline README.md:51]:
```suggestion
Packages that wrap proprietary libraries (or otherwise restrictive libraries) are
```

**Review by aviks — COMMENTED** [review]
Looks good to me, other than the one suggestion here.

File: README.md [hunk 2]
```diff
@@ -48,8 +48,11 @@ repository after the new release of your package is merged into the registry.
 
 Registered packages must have an [Open Source Initiative approved license](https://opensource.org/licenses),
 clearly marked via a `LICENSE.md`, `LICENSE`, `COPYING` or similarly named file in the package repository.
-Packages that wrap proprietary libraries are acceptable if the licenses of those libraries permit open
-source distribution of the Julia wrapper code.
+Packages that wrap proprietary libraries (or otherwise restrictive libraries) are
+acceptable if the licenses of those libraries permit open source distribution of the Julia wrapper code.
+The LICENCE file (and preferally the README) should mention the more restrictive license of the wrapped
```
Comment (ericphanson) [inline README.md:53]:
```suggestion
The LICENCE file (and preferably the README) should mention the more restrictive license of the wrapped
```

**Comment by SaschaMann** [comment]
There's a competing convention on github to put this kind of info into a different file than LICENSE, such as THIRD_PARTY_NOTICE or similar. This has some benefits, like better integrations with tooling and it being easier to see on a glance. The linked ecos repo is a good example, because its easy to assume its MIT licensed due to the start of the file and because the extra sentence is easy to miss. It's also not properly detected as an MIT licensed repo by github. 

Not arguing that a separate file like that should be required but the rules should allow for it imo.

**Comment by DilumAluthge** [comment]
Not stale

File: README.md [hunk 3]
```diff
@@ -48,8 +48,11 @@ repository after the new release of your package is merged into the registry.
 
 Registered packages must have an [Open Source Initiative approved license](https://opensource.org/licenses),
 clearly marked via a `LICENSE.md`, `LICENSE`, `COPYING` or similarly named file in the package repository.
-Packages that wrap proprietary libraries are acceptable if the licenses of those libraries permit open
-source distribution of the Julia wrapper code.
+Packages that wrap proprietary libraries (or otherwise restrictive libraries) are
+acceptable if the licenses of those libraries permit open source distribution of the Julia wrapper code.
+The LICENCE file (and preferably the README) should mention the more restrictive license of the wrapped
```
Comment (DilumAluthge) [inline README.md:53]: @SaschaMann 

```suggestion
The more restrictive license of the wrapped code:
1. MUST be mentioned in either the `THIRD_PARTY_NOTICE` file or the `LICENSE` file (preferably the `THIRD_PARTY_NOTICE` file).
3. SHOULD be mentioned in the `README` file.

Comment (SaschaMann) [reply README.md]: (Where's `2.`?) 

That addresses my concerns 👍

The file formats sometimes differ (no extension vs md etc). Given the `must`, that's something perhaps worth mentioning. But maybe that's too pedantic.

**Comment by DilumAluthge** [comment]
FWIW, I'd be fine with changing this in the future to require that these notices go in the third party notice file.

**Pull request merged**